### DISCUSSION TITLE
updateOne MongoDB collection data with array

### DISCUSSION THREAD

**Discussion by AnasTHH** [Budibase/budibase#7056] [Help] [answered]
Hi,
I was trying to store data into MongoDB that contains arrays but I couldn't.

![image](https://user-images.githubusercontent.com/15839954/182568358-8c2642cf-9240-40a5-9c8f-0cb32a1ce148.png)

However, if I limit the array size to one item one, it works fine.

![image](https://user-images.githubusercontent.com/15839954/182568647-d75be480-ee6b-4eb0-aa49-5f602c5f468c.png)

Can anyone help? is it the syntax?

**Comment by AnasTHH** [comment]
InsertOne is working
![image](https://user-images.githubusercontent.com/15839954/182764887-4efa17f2-ff32-4208-aef3-b8d9912b5464.png)

**Comment by melohagan** [comment, marked as answer]
Hi @AnasTHH Thanks for raising this, I'll work on a fix.

In the meantime I found that the following works:

```json
{
  "filter": {
	"id": "{{id}}"
  },
  "update": {
	"$set": {
		"data": {{DataArray}}
	}
  },
  "options": {
	"upsert": true
  }
}
```

**Reply by melohagan** [reply]
PR up: https://github.com/Budibase/budibase/pull/6895
